packages/connections/src/stores/connections-store.ts

```diff
@@ -351,6 +360,42 @@ export function useConnections(
           type: 'hide-store-connection-error',
         });
       },
+      async saveConnection(connectionInfo: ConnectionInfo) {
+        await saveConnectionInfo(connectionInfo);
+
+        const existingConnectionIndex = connections.findIndex(
+          (connection) => connection.id === connectionInfo.id
+        );
+
+        const newConnections = [...connections];
+
+        if (existingConnectionIndex !== -1) {
+          // Update the existing saved connection.
+          newConnections[existingConnectionIndex] = {
+            ...cloneDeep(newConnections[existingConnectionIndex]),
+            ...cloneDeep(connectionInfo),
+          };
+        } else {
+          // Add the newly saved connection to our connections list.
+          newConnections.push(cloneDeep(connectionInfo));
+        }
+
+        dispatch({
+          type: 'set-connections',
+          connections: newConnections,
+        });
+
+        // Update the active connection if it's currently selected.
+        if (activeConnectionId === connectionInfo.id) {
+          // TODO: Before merging, use the new action handler added
+          // in https://github.com/mongodb-js/compass/pull/2666/files#diff-0abd9783e312f714b88197be4cd8d7d5f84caa00fa6bce8d62d07c0c4246519eR407
```

Comment:
Will clean this up when the linked pr is merged, I think it's probably not a blocker for review.